liana-gui/src/export.rs

```diff
@@ -343,42 +345,63 @@ impl State {
     }
 }
 
-pub async fn export_subscription(mut state: State) -> (ExportProgress, State) {
-    match state.state() {
-        Status::Init => {
-            state.start().await;
-        }
-        Status::Stopped => {
-            sleep(time::Duration::from_millis(1000)).await;
-            return (ExportProgress::None, state);
-        }
-        Status::Running => { /* continue */ }
-    }
-    let msg = state.receiver.try_recv();
-    let disconnected = match msg {
-        Ok(m) => return (m, state),
-        Err(e) => match e {
-            std::sync::mpsc::TryRecvError::Empty => false,
-            std::sync::mpsc::TryRecvError::Disconnected => true,
-        },
-    };
-
-    let handle = match state.handle.take() {
-        Some(h) => h,
-        None => return (ExportProgress::Error(Error::HandleLost), state),
-    };
-    {
-        let h = handle.lock().expect("should not fail");
-        if h.is_finished() {
-            return (ExportProgress::Finished, state);
-        } else if disconnected {
-            return (ExportProgress::Error(Error::ChannelLost), state);
+pub fn export_subscription(
+    daemon: Arc<dyn Daemon + Sync + Send>,
+    path: PathBuf,
+) -> impl Stream<Item = ExportProgress> {
+    iced::stream::channel(1, move |mut output| async move {
+        let mut state = State::new(daemon, Box::new(path));
+        loop {
+            match state.state() {
+                Status::Init => {
+                    state.start().await;
+                }
+                Status::Stopped => {
+                    sleep(time::Duration::from_millis(1000)).await;
+                    let _ = output.send(ExportProgress::None).await;
```

Comment:
```suggestion
                    break;
```
this should be more clean i think, The Progress::None where an hack to deal w/ the previous subscription mechanism